### ISSUE TITLE
8.2.0 still crash

### ISSUE THREAD

**Issue by shenyinghao** [getsentry/sentry-cocoa#2769]
### Platform

iOS

### Installed

CocoaPods

### Version

8.2.0

### Steps to Reproduce

libobjc.A.dylib
objc_release_x0 + 16
1
libobjc.A.dylib
objc_release + 16
2

SentryDispatchQueueWrapper.m - 第 68 行
-[SentryDispatchQueueWrapper dispatchAfter:block:] + 68
3

SentryTracer.m - 第 223 行
-[SentryTracer dispatchIdleTimeout] + 223
4

SentryTracer.m - 第 354 行
-[SentryTracer spanFinished:] + 354
5

SentrySpan.m - 第 157 行
-[SentrySpan finishWithStatus:] + 157
6

SentryNetworkTracker.m - 第 290 行
-[SentryNetworkTracker urlSessionTask:setState:] + 290
7

SentryNetworkTrackingIntegration.m - 第 72 行
__57+[SentryNetworkTrackingIntegration swizzleURLSessionTask]_block_invoke_2.27 + 72

### Expected Result

no crash

### Actual Result

There are many crashes. Please solve this problem as soon as possible. Currently, the crashes caused by sentry SDK account for 90%

**Comment by brustolin** [comment]
Hello @shenyinghao.  Do you have a reliable way to reproduce this error? Can you share you us some code?

**Comment by shenyinghao** [comment]
> Hello @shenyinghao. Do you have a reliable way to reproduce this error? Can you share you us some code?

This is an online problem. Only online crash logs can be provided. There is no recurrence during the development process, but there are many online crashes. Please help me check it

**Comment by brustolin** [comment]
Thanks @shenyinghao.
Does Sentry captures this crash?
If so, can you send the event json to dhiogo.brustolin (at) sentry.io it would help a lot.

**Comment by shenyinghao** [comment]
0
libobjc.A.dylib
objc_retain + 16
1

SentryDispatchQueueWrapper.m - 第 59 行
-[SentryDispatchQueueWrapper dispatchAfter:block:] + 59
2

SentryTracer.m - 第 223 行
-[SentryTracer dispatchIdleTimeout] + 223
3

SentryTracer.m - 第 354 行
-[SentryTracer spanFinished:] + 354
4

SentrySpan.m - 第 157 行
-[SentrySpan finishWithStatus:] + 157
5

SentryNSDataTracker.m - 第 257 行
-[SentryNSDataTracker finishTrackingNSData:span:] + 257
6

SentryNSDataTracker.m - 第 88 行
-[SentryNSDataTracker measureNSData:writeToFile:options:error:method:] + 88
7

SentryNSDataSwizzling.m - 第 82 行
__38+[SentryNSDataSwizzling swizzleNSData]_block_invoke_2.35 + 82
8
libswiftFoundation.dylib
closure #1 in Data.write(to:options:) + 192
11
libswiftFoundation.dylib
Data.write(to:options:) + 96
12

DiskStorage.swift - 第 150 行
specialized DiskStorage.Backend.store(value:forKey:expiration:writeOptions:) + 150
13

ImageCache.swift - 第 418 行
ImageCache.syncStoreToDisk(_:forKey:processorIdentifier:callbackQueue:expiration:writeOptions:completionHandler:) + 418
14

ImageCache.swift - 第 320 行
closure #2 in ImageCache.store(_:original:forKey:options:toDisk:completionHandler:) + 320
15

<compiler-generated> - 第 1922980 行
partial apply for closure #2 in ImageCache.store(_:original:forKey:options:toDisk:completionHandler:) + 1922980
23
libsystem_pthread.dylib
start_wqthread + 8

**Comment by shenyinghao** [comment]
This is the most detailed record. Although the sentry platform has not found the report of this problem, I have this problem in the firebase-bug statistics

**Comment by shenyinghao** [comment]
I believe you have solved some problems, but this problem still exists in version 8.2.0










已崩溃：com.apple.NSURLSession-work
SIGSEGV 0x0000000b8b31a390
0
libobjc.A.dylib
objc_release + 16
1

SentryDispatchQueueWrapper.m - 第 68 行
-[SentryDispatchQueueWrapper dispatchAfter:block:] + 68
2

SentryTracer.m - 第 223 行
-[SentryTracer dispatchIdleTimeout] + 223
3

SentryTracer.m - 第 354 行
-[SentryTracer spanFinished:] + 354
4
SentrySpan.m - 第 157 行
-[SentrySpan finishWithStatus:] + 157
5

SentryNetworkTracker.m - 第 290 行
-[SentryNetworkTracker urlSessionTask:setState:] + 290
6

SentryNetworkTrackingIntegration.m - 第 72 行
__57+[SentryNetworkTrackingIntegration swizzleURLSessionTask]_block_invoke_2.27 + 72
7
CFNetwork
_CFHTTPMessageSetResponseProxyURL + 5104
23
libsystem_pthread.dylib
start_wqthread + 8

**Comment by philipphofmann** [comment]
Again @shenyinghao, please give us the error message of the crash.

**Comment by philipphofmann** [comment]
@shenyinghao, I still miss the error message. An example would be 
```
EXC_BAD_ACCESS
crash > crash: > dyld_v1  arm64e > objectAtIndex: >
Attempted to dereference null pointer.
```

**Comment by wxhzzsf** [comment]
One example would be:

> Crashed:com.onevcat.Kingfisher.ImageCache.ioQueue.E9E8B081-6F5D-456A-ACDC-0564F1830074
> SIGSEGV 0x0000000000000008

**Comment by wxhzzsf** [comment]
Here is another Example:
```
Crashed: com.google. GDTCORFlatFileStorage
SIGSEGV 0x0000000000000008
```

**Comment by philipphofmann** [comment]
@wxhzzsf, I don't understand why you think your error messages relate to this crash, as they don't point to Sentry. The stacktraces of @shenyinghao point all to the following location.

```
1 0xe94e00 -[SentryDispatchQueueWrapper dispatchAfter:block:] + 68 (SentryDispatchQueueWrapper.m:68)
2 0xecd98c -[SentryTracer dispatchIdleTimeout] + 223 (SentryTracer.m:223)
3 0xece620 -[SentryTracer spanFinished:] + 354 (SentryTracer.m:354)
4 0xec66c0 -[SentrySpan finishWithStatus:] + 157 (SentrySpan.m:157)
5 0xeaa864 -[SentryNetworkTracker urlSessionTask:setState:] + 290 (SentryNetworkTracker.m:290)
6 0xeabfe8 __57+[SentryNetworkTrackingIntegration swizzleURLSessionTask]_block_invoke_2.27 + 72 (SentryNetworkTrackingIntegration.m:72)
```

Can you please share some stacktraces, including the error message, @wxhzzsf?

**Comment by wxhzzsf** [comment]
```
Crashed: com.google.GDTCORFlatFileStorage
SIGSEGV 0x0000da0215259a50
```
```
Crashed: com.google.GDTCORFlatFileStorage
0  libobjc.A.dylib                0x1c20 objc_msgSend + 32
1  libobjc.A.dylib                0x1cb30 objc_object::sidetable_release(bool, bool) + 292
2  CIDER                          0xe9e360 -[SentryDispatchQueueWrapper dispatchAfter:block:] + 68 (SentryDispatchQueueWrapper.m:68)
3  CIDER                          0xed8008 -[SentryTracer dispatchIdleTimeout] + 228 (SentryTracer.m:228)
4  CIDER                          0xed8cc8 -[SentryTracer spanFinished:] + 360 (SentryTracer.m:360)
5  CIDER                          0xed0eac -[SentrySpan finishWithStatus:] + 156 (SentrySpan.m:156)
6  CIDER                          0xeb82fc -[SentryNSDataTracker finishTrackingNSData:span:] + 257 (SentryNSDataTracker.m:257)
7  CIDER                          0xeb7604 -[SentryNSDataTracker measureNSData:writeToFile:options:error:method:] + 88 (SentryNSDataTracker.m:88)
8  CIDER                          0xeb6c80 __38+[SentryNSDataSwizzling swizzleNSData]_block_invoke_2.35 + 82 (SentryNSDataSwizzling.m:82)
9  CIDER                          0x87eb0 GDTCORWriteDataToFile + 295 (GDTCORPlatform.m:295)
10 CIDER                          0xbef5b0 __47-[GDTCORFlatFileStorage storeEvent:onComplete:]_block_invoke.42 + 171 (GDTCORFlatFileStorage.m:171)
11 libdispatch.dylib              0x2460 _dispatch_call_block_and_release + 32
12 libdispatch.dylib              0x3f88 _dispatch_client_callout + 20
13 libdispatch.dylib              0xb640 _dispatch_lane_serial_drain + 672
14 libdispatch.dylib              0xc18c _dispatch_lane_invoke + 384
15 libdispatch.dylib              0x16e10 _dispatch_workloop_worker_thread + 652
16 libsystem_pthread.dylib        0xdf8 _pthread_wqthread + 288
17 libsystem_pthread.dylib        0xb98 start_wqthread + 8
```

**Comment by wxhzzsf** [comment]
```
Crashed:com.onevcat.Kingfisher.ImageCache.ioQueue.DB207702-CEFF-4A61-9858-B90C4CE326AF
SIGSEGV 0x000000095e06cc60
```
```
Crashed: com.onevcat.Kingfisher.ImageCache.ioQueue.DB207702-CEFF-4A61-9858-B90C4CE326AF
0  libobjc.A.dylib                0x3554 objc_release_x0 + 16
1  libobjc.A.dylib                0x3554 objc_release + 16
2  CIDER                          0xe801bc -[SentryDispatchQueueWrapper dispatchAfter:block:] + 68 (SentryDispatchQueueWrapper.m:68)
3  CIDER                          0xeb8c6c -[SentryTracer dispatchIdleTimeout] + 223 (SentryTracer.m:223)
4  CIDER                          0xeb9760 -[SentryTracer spanFinished:] + 344 (SentryTracer.m:344)
5  CIDER                          0xeb1abc -[SentrySpan finishWithStatus:] + 149 (SentrySpan.m:149)
6  CIDER                          0xe994cc -[SentryNSDataTracker finishTrackingNSData:span:] + 257 (SentryNSDataTracker.m:257)
7  CIDER                          0xe987d4 -[SentryNSDataTracker measureNSData:writeToFile:options:error:method:] + 88 (SentryNSDataTracker.m:88)
8  CIDER                          0xe97e50 __38+[SentryNSDataSwizzling swizzleNSData]_block_invoke_2.35 + 82 (SentryNSDataSwizzling.m:82)
9  Foundation                     0x417b9c specialized __DataStorage.withInteriorPointerReference<A>(_:_:) + 192
10 Foundation                     0x420b70 specialized Data._Representation.withInteriorPointerReference<A>(_:) + 432
11 Foundation                     0x424b08 Data.write(to:options:) + 68
12 CIDER                          0xc7cc84 specialized DiskStorage.Backend.store(value:forKey:expiration:writeOptions:) + 150 (DiskStorage.swift:150)
13 CIDER                          0xc813cc ImageCache.syncStoreToDisk(_:forKey:processorIdentifier:callbackQueue:expiration:writeOptions:completionHandler:) + 418 (ImageCache.swift:418)
14 CIDER                          0xc80e10 closure #2 in ImageCache.store(_:original:forKey:options:toDisk:completionHandler:) + 320 (ImageCache.swift:320)
15 CIDER                          0xc888c0 partial apply for closure #2 in ImageCache.store(_:original:forKey:options:toDisk:completionHandler:) + 1862616 (<compiler-generated>:1862616)
16 CIDER                          0x11070 thunk for @escaping @callee_guaranteed () -> () + 4329689200 (<compiler-generated>:4329689200)
17 libdispatch.dylib              0x2460 _dispatch_call_block_and_release + 32
18 libdispatch.dylib              0x3f88 _dispatch_client_callout + 20
19 libdispatch.dylib              0xb640 _dispatch_lane_serial_drain + 672
20 libdispatch.dylib              0xc18c _dispatch_lane_invoke + 384
21 libdispatch.dylib              0x16e10 _dispatch_workloop_worker_thread + 652
22 libsystem_pthread.dylib        0xdf8 _pthread_wqthread + 288
23 libsystem_pthread.dylib        0xb98 start_wqthread + 8
```

**Comment by philipphofmann** [comment]
Fixed with #2834
